crates/rome_formatter/src/lib.rs

```diff
@@ -76,7 +78,23 @@ use std::str::FromStr;
 
 /// This trait should be implemented on each node/value that should have a formatted representation
 pub trait ToFormatElement {
-	fn to_format_element(&self, formatter: &Formatter) -> Option<FormatElement>;
+	fn to_format_element(&self, formatter: &Formatter) -> Result<FormatElement, FormatError>;
+}
+
+#[derive(Debug, PartialEq)]
+/// Series of errors encountered during formatting
+pub enum FormatError {
+	/// Node is missing and it should be required for a correct formatting
+	MissingNode(SyntaxKind),
```

Comment:
```suggestion
	MissingElement(SyntaxKind),
```

It can be either a token or node that is missing. 

There's also a problem with passing in the missing kind in case the missing child is a union, e.g. an expression. There isn't a single missing kind in that case. 

We should, therefore, either remove the parameter or re-define the error. An alternative that I see is to change it to `MissingRequiredChild { parent: SyntaxNode }`. We may be able to add the "slot" of the missing child in the future so that it is clear which child was missing.